src/tracking.F90

```diff
@@ -92,6 +92,9 @@ subroutine transport(p)
         ! Since the MGXS can be angle dependent, this needs to be done
         ! After every collision for the MGXS mode
         if (p % material /= MATERIAL_VOID) then
+          ! Update the temperature index
+          call macro_xs(p % material) % obj % find_temperature(p % sqrtkT)
+          ! Get the data
           call macro_xs(p % material) % obj % calculate_xs(p % g, &
                p % coord(p % n_coord) % uvw, material_xs)
         else
```

Comment:
This comment was from tracking.F90 which isn't what you intended.  Is there an mgxs.h5 file that is in version control from the testing folder?